internal/key/key.go

```diff
@@ -12,27 +12,16 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-package provider
+package key
```

Comment:
Can we add a few lines how this package helps us figure out what is available in our context?

Reply:
I can add a few lines detailing why the package exists:

```go
// key provides an internal set of keys for use with [context.WithValue] and
// [context.Context.Value] that can be shared across packages source.
//
// Each key has a private type (a `struct{}`) and a public instance of that type.
```

> Can we add a few lines how this package helps us figure out what is available in our context?

`keys` doesn't help us figure out what is available in our context. It exists purely so we can share keys between packages in pulumi-go-provider without leaking that implementation to our users.